src/error.rs

```diff
@@ -25,6 +25,9 @@ pub enum Error<E: Debug + Display> {
     #[error("Could not determine app directory: {0}")]
     CannotDetermineAppDirectory(std::io::Error),
 
+    #[error("{0}")]
+    Other(String),
```

Comment:
This feels like an anti-pattern to me, especially when we introduce this for a _specific_ feature. I'm not saying we should do this, but even naming this `DetectContextBuilderError(String)` would be miles better. 

Ideally, we don't have this at all.

Reply:
Agreed, this is very much *bad code* ™️ that I needed to get my code to compile.

Reply:
I updated the error handling, please take a look.